apps/fishing-map/features/datasets/upload/NewTrackDataset.tsx

```diff
@@ -156,22 +153,31 @@ function NewTrackDataset({
         ...(meta as DatasetMetadata),
         configuration: {
           ...meta?.configuration,
-          ...(newConfig as DatasetMetadata['configuration']),
+          configurationUI: {
+            ...meta?.configuration?.configurationUI,
+            ...(newConfig as DatasetMetadata['configuration']),
+          },
         },
       }))
     },
     []
   )
 
   const fieldsOptions: SelectOption[] = useMemo(
-    () => datasetMetadata?.fields?.map((field) => ({ id: field, label: field })) || [],
+    () =>
+      (datasetMetadata?.schema &&
+        Object.keys(datasetMetadata.schema).map((field) => ({ id: field, label: field }))) ||
+      [],
     [datasetMetadata]
   )
 
   const selectedOption = useCallback(
     (option: string): SelectOption => ({
-      label: datasetMetadata?.configuration?.[option] as string,
-      id: datasetMetadata?.configuration?.[option],
+      label: (datasetMetadata?.configuration?.configurationUI?.[option] ||
```

Comment:
Instead of generating a new option with its possible error or keeping it unsynced with the original options, why not do something like:
```
fieldsOptions.find(o => o.id === option)
```

Reply:
But as far as I understand it what we want to display as `selectedOption` is the one that is currently selected in the dataset configuration